### ISSUE TITLE
Add GraalVM support to prometheus-rsocket-proxy

### ISSUE THREAD

**Issue by klopfdreh** [micrometer-metrics/prometheus-rsocket-proxy#85]
Because `prometheus-rsocket-proxy` uses `snappy-java` and because this referenced library makes use of native libs packed into the dependency, the following PR needs to be merged so that GraalVM is supported: https://github.com/xerial/snappy-java/pull/590. `snappy-java` is used in the submodule `client`, `proxy` and `proxy-server`

See: 
* https://github.com/micrometer-metrics/prometheus-rsocket-proxy/blob/main/client/build.gradle#L12
* https://github.com/micrometer-metrics/prometheus-rsocket-proxy/blob/main/proxy-server/build.gradle#L28
* https://github.com/micrometer-metrics/prometheus-rsocket-proxy/blob/main/proxy/build.gradle#L33

**Comment by shakuzen** [comment]
I'm not sure there's anything we can/should do about this other than upgrade the version of snappy-java when/if a version is released that supports GraalVM. Am I missing anything?

**Comment by klopfdreh** [comment]
@shakuzen - this is right. I just added the issue here as others are also facing it. If `snappy-java` is released with the referenced PR you can update the dependency and close this issue. 👍

**Comment by shakuzen** [comment]
Thanks for the clarification and making us aware. We'll wait until we can upgrade.

**Comment by klopfdreh** [comment]
New version is released: https://github.com/xerial/snappy-java/releases/tag/v1.1.10.6 - but I guess we have to wait a bit until it is available at maven central.

**Comment by shakuzen** [comment]
Well that was quite a quick turnaround 😄

**Comment by shakuzen** [comment]
Could you confirm that the latest snapshots work with GraalVM native images now that the new version of snappy-java should be pulled in?

**Comment by klopfdreh** [comment]
I tested it already for snappy-java - if the version is resolved correctly via dependency management I think we are safe. 👍

**Comment by shakuzen** [comment]
Dependency management in this project is currently done with dynamic versions and lock files for release versions. `main` currently doesn't have any lock files, so the latest version of snappy-java should have been picked up in the latest snapshot build. I think we can close this if the latest snapshots are able to work with GraalVM native image compilation and running.

**Comment by klopfdreh** [comment]
Okay then let me check it - are the Snapshots also available in maven central?

**Comment by shakuzen** [comment]
Snapshots are in https://repo.spring.io/snapshot. See https://github.com/micrometer-metrics/micrometer?tab=readme-ov-file#snapshot-builds

**Comment by klopfdreh** [comment]
Hey @shakuzen - I created a demo project for you so that you are also able to test native support in future: https://github.com/klopfdreh/prometheus-rsocket-native-test

The README.md contains all required information what to do to create a Spring Boot application with prometheus-rsocket-proxy-client in a native docker image.

The server is not native compiled yet.

All in all - long explanation short answer: **It works** 😄 

I used `2.0.0-SNAPSHOT` and the newest `snappy-java` version was provided as dependency.
